lightning/src/ln/channel.rs

```diff
@@ -4568,6 +4635,9 @@ impl<Signer: Sign> Writeable for Channel<Signer> {
 		self.counterparty_shutdown_scriptpubkey.write(writer)?;
 
 		self.commitment_secrets.write(writer)?;
+
+		write_tlv_fields!(writer, {}, {(0, self.announcement_sigs)});
```

Comment:
Think it'd be worth adding a serialization round-trip for this to a test? I guess I'm feeling the pressure to be correct on this PR.. Also maybe a unit test serialization round-trip with multiple TLV fields.

Reply:
Hmm, we already have generic serialization roundtrip tests which get run after every functional test, so I think we have pretty good coverage here :). I went ahead and added some tests for the new optional field stuff in `ser_macros.rs` as well (which this is a thing wrapper around).

Reply:
Done.